src/actions/gifs.js

```diff
@@ -0,0 +1,604 @@
+    };
+}
+
+export function analyticsEvent({event, params}) {
+    return {
+        type: GifTypes.SEND_EVENT,
+        payload: {
+            event,
+            params,
+        },
+    };
+}
+
+export function setUserIdCookie(utc) {
+    return {
+        type: GifTypes.SET_USER_COOKIE,
+        payload: {
+            utc,
+        },
+    };
+}
+
+export function setSessionIdCookie(stc) {
+    return {
+        type: GifTypes.SET_SESSION_COOKIE,
+        payload: {
+            stc,
+        },
+    };
+}
+
+export function sendViewCount({gfyId, params}) {
+    return (dispatch, getState) => {
+        if (typeof document === 'undefined') {
+            return;
+        }
+        if (!params.app_id) {
+            params.add_id = 'com.gfycat.website';
+        }
+
+        dispatch(viewCount({gfyId, params}));
+        const {utc, stc} = getUserAndSessionIds({dispatch, getState});
+        const encodedUtc = encodeURIComponent(utc);
+        const encodedStc = encodeURIComponent(stc);
+        const deviceType = 'unknown';
+
+        const extendedParams = {
+            gfyid: gfyId,
+            ...params,
+            device_type: deviceType,
+            utc: encodedUtc,
+            stc: encodedStc,
+            rand: Math.random() * 100000,
+        };
+
+        const queryString = queryParamsString(extendedParams);
+        const url = `https://px.gfycat.com/px.gif?${queryString}`;
+        fetch(url);
+    };
+}
+
+export function sendViewCountBatch({gfycats, params}) {
+    return (dispatch, getState) => {
+        if (typeof document === 'undefined') {
+            return;
+        }
+        if (!params.app_id) {
+            params.add_id = 'com.gfycat.website';
+        }
+
+        const {utc, stc} = getUserAndSessionIds({dispatch, getState});
+        const encodedUtc = encodeURIComponent(utc);
+        const encodedStc = encodeURIComponent(stc);
+
+        if (gfycats.length > BATCH_MAX_LENGTH) {
+            const chunks = [];
+            while (gfycats.length) {
+                chunks.push(gfycats.splice(0, BATCH_MAX_LENGTH));
+            }
+            for (let i = 0; i < chunks.length; i++) {
+                dispatch(sendViewCountBatch({gfycats: chunks[i], params}));
+            }
+            return;
+        }
+
+        const gfyIds = gfycats.reduce((ids, gfyItem, index) => {
+            const key = index === 0 ? 'gfyid' : `gfyid_${index}`;
+            ids[key] = gfyItem.gfyId;
+            return ids;
+        }, {});
+
+        const extendedParams = {
+            ...params,
+            ...gfyIds,
+            utc: encodedUtc,
+            stc: encodedStc,
+        };
+
+        const queryString = queryParamsString(extendedParams);
+
+        const url = `https://px.gfycat.com/px.gif?${queryString}`;
+        fetch(url);
+    };
+}
+
+export function splitBatchRequest({gfycats, params, dispatch}) {
+    dispatch(sendViewCountBatch({gfycats, params}));
+}
+
+export function sendEvent({event, params}) {
```

Comment:
I understand this was discussed at the design level, but I'm concerned by the event tracking here.

1) Is this required to be allowed to use the Gfycat API? (If so, I think this lends credence to needing to put the onus on the enduser to acquire a Gfycat credentials as asked below.)
2) If required, but the end user is required to configure this, is it required to report metrics at the granularity designed here? I'm concerned that this would unnecessarily expose user count/engagement. Is it sufficient to ping the metrics endpoint instead (recognizing that IP address leakage still occurs)?

Reply:
Good catch. @kennethjeremyau We have pretty strict agreements with our users on what tracking is allowed, and what data is collected - if any.

We'll need to tie this with our [Diagnostics and Error Reporting feature](https://docs.mattermost.com/administration/telemetry.html#error-and-diagnostics-reporting-feature) if we want to collect data.

Reply:
I've removed our analytics and am using `client.trackEvent()`.